addons/account_payment/controllers/portal.py

```diff
@@ -15,35 +18,104 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
+        total_amount = sum(overdue_invoices.mapped('amount_total'))
+        residual_amount = sum(overdue_invoices.mapped('amount_residual'))
+        payment_date = fields.Date.today()
+        batch_name = company.get_next_batch_payment_communication(payment_date, overdue_invoices)
+        values['payment'] = {
+            'date': payment_date,
+            'reference': batch_name,
+            'amount': total_amount,
+            'currency': overdue_invoices[0].company_currency_id,
+        }
+
+        common_view_values = self._get_common_page_view_values(
+            invoices_data={
+                'partner': partner,
+                'company': company,
+                'total_amount': total_amount,
+                'currency': currency,
+                'residual_amount': residual_amount,
+                'invoice_ids': overdue_invoices.mapped('id'),
+                'payment_reference': batch_name,
+                'landing_route': '/my/invoices/',
+            },
+            access_token=access_token,
+            **kwargs)
+        values |= common_view_values
+        return values
+
+    def _get_common_page_view_values(self, invoices_data, access_token, **kwargs):
         logged_in = not request.env.user._is_public()
         # We set partner_id to the partner id of the current user if logged in, otherwise we set it
         # to the invoice partner id. We do this to ensure that payment tokens are assigned to the
         # correct partner and to avoid linking tokens to the public user.
-        partner_sudo = request.env.user.partner_id if logged_in else invoice.partner_id
-        invoice_company = invoice.company_id or request.env.company
+        partner_sudo = request.env.user.partner_id if logged_in else invoices_data['partner']
+        invoice_company = invoices_data['company'] or request.env.company
 
         availability_report = {}
         # Select all the payment methods and tokens that match the payment context.
         providers_sudo = request.env['payment.provider'].sudo()._get_compatible_providers(
             invoice_company.id,
             partner_sudo.id,
-            invoice.amount_total,
-            currency_id=invoice.currency_id.id,
+            invoices_data['total_amount'],
+            currency_id=invoices_data['currency'].id,
             report=availability_report,
         )  # In sudo mode to read the fields of providers and partner (if logged out).
         payment_methods_sudo = request.env['payment.method'].sudo()._get_compatible_payment_methods(
             providers_sudo.ids,
             partner_sudo.id,
-            currency_id=invoice.currency_id.id,
+            currency_id=invoices_data['currency'].id,
             report=availability_report,
         )  # In sudo mode to read the fields of providers.
         tokens_sudo = request.env['payment.token'].sudo()._get_available_tokens(
-            providers_sudo.ids, partner_sudo.id
+            providers_sudo.ids, partner_sudo.id,
```

Comment:
is ruff complaining for that ? otherwise i would not change even if i prefer the comma at the end

Reply:
nope did not complain